core-feature-pack/pom.xml

```diff
@@ -112,6 +112,11 @@
 
         <dependency>
             <groupId>org.jboss.logging</groupId>
+            <artifactId>jboss-logging-annotations</artifactId>
```

Comment:
As kabir says, this is only used for compile time generation of logging interface / classes.
It probably got into feature-pack module by mistake.